bugsnag-plugin-android-okhttp/src/main/java/com/bugsnag/android/okhttp/OkHttpDelivery.kt

```diff
@@ -0,0 +1,100 @@
+package com.bugsnag.android.okhttp
+
+import android.annotation.SuppressLint
+import com.bugsnag.android.Configuration
+import com.bugsnag.android.Delivery
+import com.bugsnag.android.DeliveryParams
+import com.bugsnag.android.DeliveryStatus
+import com.bugsnag.android.EventPayload
+import com.bugsnag.android.Logger
+import com.bugsnag.android.Session
+import com.bugsnag.android.internal.DeliveryHelper.getDeliveryStatus
+import com.bugsnag.android.internal.DeliveryHelper.retrieveSha1Digest
+import com.bugsnag.android.internal.DeliveryHelper.serializePayload
+import com.bugsnag.android.internal.JsonHelper
+import okhttp3.Headers
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.io.IOException
+
+class OkHttpDelivery constructor(
+    private val existingOkHttpClient: OkHttpClient,
+    private val apiKey: String,
+    private val maxStringValueLength: Int,
+    private val logger: Logger
+) : Delivery {
+
+    companion object {
+        @SuppressLint("NotConstructor")
+        @JvmSynthetic
+        fun Configuration.OkHttpDelivery(okHttpClient: OkHttpClient): OkHttpDelivery {
+            return OkHttpDelivery(okHttpClient, apiKey, maxStringValueLength, logger!!)
+        }
+
+        fun Configuration.useOkHttpDelivery(okHttpClient: OkHttpClient) {
+            delivery = OkHttpDelivery(okHttpClient)
+        }
```

Comment:
These should be outside of the `OkHttpDelivery` class (ie: in the global namespace).